src/components/ui/ImageGallery/ImageGallery.tsx

```diff
@@ -0,0 +1,58 @@
+import { Button } from '@faststore/ui'
+import React, { useState } from 'react'
+import { Image } from 'src/components/ui/Image'
+
+import { ImageGallerySelector, ImageZoom } from '.'
+
+export interface ImageElementData {
+  url: string
+  alternateName: string
+}
+
+interface ImageGalleryProps {
+  images: ImageElementData[]
+}
+
+function ImageGallery({ images }: ImageGalleryProps) {
+  const [selectedImageIdx, setSelectedImageIdx] = useState(0)
+  const currentImage = images[selectedImageIdx]
+
+  return (
+    <div>
+      <ImageZoom>
+        <Image
+          src={currentImage.url}
+          alt={currentImage.alternateName}
+          variant="product.details"
+          loading="eager"
+        />
+      </ImageZoom>
+      <ImageGallerySelector itemsPerPage={4}>
+        {images.map((image, idx) => {
+          return (
+            <Button
+              key={idx}
+              data-thumbnail-button
+              className={image.url === currentImage.url ? 'selected' : ''}
```

Comment:
Hmm, I don't know. I'd rather stay with the 'selected' in the classname rather than using the data attribute for that. Personally I'd prefer doing `[data-thumbnail-button].selected` instead of `[data-thumbnail-button='selected']`. On both cases, the 'default' button state would be styled with just `[data-thumbnail-button]`, right?

Not sure which way is better. @renatamottam, @saranicoly - thoughts on this?

Reply:
Let's go with igor's suggestion! Can you please use `[data-thumbnail-button='selected']` instead? We are trying to follow this pattern for the other components too :)

Reply:
Great! So data attribute it is. Thanks, @igorbrasileiro

Reply:
Done at https://github.com/vtex-sites/base.store/pull/142/commits/11880b496ee56279d428dca98d447ac477542b58